planning.ipynb

```diff
@@ -302,13 +303,185 @@
+       "    <span class=\"n\">precond_neg</span> <span class=\"o\">=</span> <span class=\"p\">[]</span>\n",
+       "    <span class=\"n\">effect_add</span> <span class=\"o\">=</span> <span class=\"p\">[</span><span class=\"n\">expr</span><span class=\"p\">(</span><span class=\"s2\">&quot;At(p, to)&quot;</span><span class=\"p\">)]</span>\n",
+       "    <span class=\"n\">effect_rem</span> <span class=\"o\">=</span> <span class=\"p\">[</span><span class=\"n\">expr</span><span class=\"p\">(</span><span class=\"s2\">&quot;At(p, f)&quot;</span><span class=\"p\">)]</span>\n",
+       "    <span class=\"n\">fly</span> <span class=\"o\">=</span> <span class=\"n\">Action</span><span class=\"p\">(</span><span class=\"n\">expr</span><span class=\"p\">(</span><span class=\"s2\">&quot;Fly(p, f, to)&quot;</span><span class=\"p\">),</span> <span class=\"p\">[</span><span class=\"n\">precond_pos</span><span class=\"p\">,</span> <span class=\"n\">precond_neg</span><span class=\"p\">],</span> <span class=\"p\">[</span><span class=\"n\">effect_add</span><span class=\"p\">,</span> <span class=\"n\">effect_rem</span><span class=\"p\">])</span>\n",
+       "\n",
+       "    <span class=\"k\">return</span> <span class=\"n\">PDDL</span><span class=\"p\">(</span><span class=\"n\">init</span><span class=\"p\">,</span> <span class=\"p\">[</span><span class=\"n\">load</span><span class=\"p\">,</span> <span class=\"n\">unload</span><span class=\"p\">,</span> <span class=\"n\">fly</span><span class=\"p\">],</span> <span class=\"n\">goal_test</span><span class=\"p\">)</span>\n",
+       "</pre></div>\n",
+       "</body>\n",
+       "</html>\n"
+      ],
+      "text/plain": [
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "psource (air_cargo)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**At(x, a):** The cargo or plane **' x '** is at airport **' a '**.\n",
+    "\n",
+    "**In(c, p):** Cargo **' c '** is in palne **' p '**.\n",
+    "\n",
+    "**Cargo(x):** Declare **' x '** as cargo.\n",
+    "\n",
+    "**Plane(x):** Declare **' x '** as plane.\n",
+    "\n",
+    "**Airport(x):** Declare **' x '** as airport.\n",
+    "\n",
+    "\n",
+    "\n",
+    "we have an `initial_state` of having cargo C1, plane P1 at airport SFO and cargo C2, plane P2 at airport JFK. Our goal state is to have cargo C1 at airport JFK and cargo C2 at airport SFO. We will discuss on how to achieve this. Let us now define an object of `air_cargo` problem:"
```

Comment:
The first sentence is a bit awkward, there are two 'have's in there. I suggest changing it to something like this:

"In the `initial_state` we have cargo C1, plane P1 ..."

I think this reads better.